pkg/storage/pv_test.go

```diff
@@ -0,0 +1,211 @@
+package storage
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+const defaultPersistentVolumeName = "persistentvolume-test"
+
+func TestPullPersistentVolume(t *testing.T) {
+	testCases := []struct {
+		persistentVolumeName string
+		addToRuntimeObjects  bool
+		client               bool
+		expectedErrorText    string
+	}{
+		{
+			persistentVolumeName: defaultPersistentVolumeName,
+			addToRuntimeObjects:  true,
+			client:               false,
+			expectedErrorText:    fmt.Sprintf("PersistentVolume object %s does not exist", defaultPersistentVolumeName),
+		},
+		{
+			persistentVolumeName: defaultPersistentVolumeName,
+			addToRuntimeObjects:  true,
+			client:               true,
+			expectedErrorText:    "",
+		},
+		{
+			persistentVolumeName: defaultPersistentVolumeName,
+			addToRuntimeObjects:  false,
+			client:               true,
+			expectedErrorText:    fmt.Sprintf("PersistentVolume object %s does not exist", defaultPersistentVolumeName),
+		},
+		{
+			persistentVolumeName: "",
+			addToRuntimeObjects:  true,
+			client:               true,
+			expectedErrorText:    "",
```

Comment:
We have to validate that persistentVolumeName is not empty string. Please add validation to the original function